avocado/core/tree.py

```diff
@@ -104,6 +104,10 @@ def __eq__(self, other):
                     return False
             return True
 
+    def __ne__(self, other):
+        """ Inverted eq """
+        return not self == other
```

Comment:
It does.  I added a breakpoint to `__ne__` and it gets called from those asserts.